### PR TITLE
Use PyUpgrade to use Python 3.6 features

### PR DESCRIPTION
**Pull request by kaxil**
Use features like `f-strings` instead of format across the code-base.

More details: https://github.com/asottile/pyupgrade

**Features**:
- Set literals
- Dictionary comprehensions
- printf-style string formatting
- Unicode literals
- Invalid escape sequences
- `.encode()` to bytes literals
- `super()` calls
- "new style" classes
- `# coding: ...` comment
- Remove unnecessary py3-compat imports
- rewrite `mock` imports
- if `PY2` blocks
- remove `six` compatibility code
- `open` alias
- OSError aliases
- `typing.NamedTuple` / `typing.TypedDict` py36+ syntax
- remove parentheses from @functools.lru_cache()
- f-strings

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by kaxil** [comment]
@turbaszek Since backport packages are Python 3.6+ too, so upgrading them is fine too, right?

**Comment by turbaszek** [comment]
> @turbaszek Since backport packages are Python 3.6+ too, so upgrading them is fine too, right?

Yes, but I'm afraid this change may cause a lot of merge conflict with currently open PRs. I think we should wait with it for 2.0-alpha, probably apply it at the same moment as black

**Comment by kaxil** [comment]
> > @turbaszek Since backport packages are Python 3.6+ too, so upgrading them is fine too, right?
> 
> Yes, but I'm afraid this change may cause a lot of merge conflict with currently open PRs. I think we should wait with it for 2.0-alpha, probably apply it at the same moment as black

Yup, this PR can wait :)

File: airflow/cli/commands/dag_command.py [hunk 1]
```diff
@@ -112,10 +112,10 @@ def dag_backfill(args, dag=None):
         run_conf = json.loads(args.conf)
 
     if args.dry_run:
-        print("Dry run of DAG {0} on {1}".format(args.dag_id,
+        print("Dry run of DAG {} on {}".format(args.dag_id,
                                                  args.start_date))
```
Comment (mik-laj) [inline airflow/cli/commands/dag_command.py]:
```suggestion
                                               args.start_date))
```

**Comment by kaxil** [comment]
Rebased, so that we can merge it before beta1

**Comment by kaxil** [comment]
@turbaszek This is ready for review, as soon as this is merged. I will quickly rebased and make https://github.com/apache/airflow/pull/9550 pass the test too

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -177,6 +177,11 @@ repos:
       - id: fix-encoding-pragma
         args:
           - --remove
+  - repo: https://github.com/asottile/pyupgrade
+    rev: v2.7.2
```
Comment (XD-DENG) [inline .pre-commit-config.yaml]: LGTM.

A minor question: shall we use the latest v2.7.3?

**Review by XD-DENG — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (kaxil) [reply .pre-commit-config.yaml]: Good call, will update

**Pull request merged**